spec/ruby/optional/capi/string_spec.rb

```diff
@@ -1236,6 +1236,14 @@ def inspect
     it "returns the same string as String#-@" do
       @s.rb_enc_interned_str_cstr("hello", Encoding::UTF_8).should.equal?(-"hello")
     end
+
+    ruby_bug "#20322", ""..."3.4" do
+      it "uses the default encoding if encoding is null" do
+        str = "hello"
+        val = @s.rb_enc_interned_str_cstr(str, nil)
+        val.encoding.should == Encoding::ASCII_8BIT
+      end
+    end
```

Comment:
This test was also added [here](https://github.com/ruby/ruby/pull/10169)—should I sync `mspec` or wait for it to be synced?

Reply:
It seems OK to commit the test here in this PR.